### PR TITLE
Fix a range of values for parameters of the Time#change

### PR DESCRIPTION
**Pull request by nkondratyev**
Passing 999999000 < `:nsec` < 999999999 and 999999 < `:usec` < 1000000
to change a time with utc_offset doesn't throw an `ArgumentError`.

These calls leads to the error:

``` ruby
Time.new(2015, 5, 9, 10, 00, 00, '+03:00').change(nsec: 999999999) # => ArgumentError: argument out of range
Time.new(2015, 5, 9, 10, 00, 00, '+03:00').change(usec: Rational(999999999, 1000)) # => ArgumentError: argument out of range
```

### CODE REVIEW

File: activesupport/test/core_ext/time_ext_test.rb [hunk 1]
```diff
@@ -149,6 +149,9 @@ def test_end_of_day
       assert_equal Time.local(2006,3,19,23,59,59,Rational(999999999, 1000)), Time.local(2006,3,19,10,10,10).end_of_day, 'ends DST'
       assert_equal Time.local(2006,10,1,23,59,59,Rational(999999999, 1000)), Time.local(2006,10,1,10,10,10).end_of_day, 'start DST'
     end
+    with_env_tz 'Asia/Yekaterinburg' do
+      assert_equal Time.local(2015, 2, 8, 23, 59, 59, Rational(999999999, 1000)), Time.new(2015, 2, 8, 8, 0, 0, '+05:00').end_of_day
+    end
```
Comment (hundredwatt) [inline activesupport/test/core_ext/time_ext_test.rb]: Is this test relevant to this the code change in this PR?

Comment (nkondratyev) [inline activesupport/test/core_ext/time_ext_test.rb]: Indirectly. All methods like `Time#end_of_*` that use `Time#change` pass `usec: Rational(999999999, 1000)` as their parameter may be affected by this error.

For example:

```ruby
Time.new(2015, 5, 9, 10, 00, 00, '+03:00').end_of_day # raise ArgumentError: argument out of range
```
